### DISCUSSION TITLE
Is there a way to disable diagnostics for a particular method?

### DISCUSSION THREAD

**Discussion by baynezy** [riok/mapperly#583] [Q&A] [answered]
I have [enabled diagnostics](https://mapperly.riok.app/docs/configuration/mapper#strict-property-mappings) as indicated in the documentation.

```
[*.cs]
dotnet_diagnostic.RMG012.severity = error # Unmapped target member
dotnet_diagnostic.RMG020.severity = error # Unmapped source member
```

This works as expected. Is there a way to disable this at a method level? My goal is that I turn it on at a global level, but indicate that I am fine if it is not processed for a particular method.

**Comment by latonz** [comment, marked as answer]
You should be able to do so with [preprocessor directives](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/suppress-warnings#use-a-preprocessor-directive).

```c#
#pragma warning disable RMG020
// ... method ...
#pragma warning restore RMG020
```

**Reply by baynezy** [reply]
That worked a treat. Thank you.
